### DISCUSSION TITLE
Upgrading from 4.3 to 10.x

### DISCUSSION THREAD

**Discussion by nirinchev** [realm/realm-dotnet#2299] [Q&A] [answered]
I was wondering if I update Realm from 4.3 to 10, will my db continue working fine? And what else can break?

_Originally posted by @rodion-m in https://github.com/realm/realm-dotnet/discussions/2158#discussioncomment-495218_

**Comment by nirinchev** [comment, marked as answer]
The [Changelog](https://github.com/realm/realm-dotnet/blob/v10/CHANGELOG.md) should contain a detailed list of all changes. In terms of breaking changes between 4.3 to 10.x, they are mostly sync-related. If you're using the local database only, the only API breaking change is moving the dynamic API moved from Realm to Realm.Dynamic. Obviously, there are many major changes under the hood, but they should not cause your app to break.

**Reply by rodion-m** [reply]
> The [Changelog](https://github.com/realm/realm-dotnet/blob/v10/CHANGELOG.md) should contain a detailed list of all changes. In terms of breaking changes between 4.3 to 10.x, they are mostly sync-related. If you're using the local database only, the only API breaking change is moving the dynamic API moved from Realm to Realm.Dynamic. Obviously, there are many major changes under the hood, but they should not cause your app to break.

Ok, great! And thanks for a fast reply. Is 10.x has many bug fixes, right? So, we should not delay updating?

**Reply by nirinchev** [reply]
10.x includes a major rewrite of the Core database which should make it perform better in certain scenarios. It adds support for new data types, such as decimal and embedded objects, with Set and Dictionary nearing completion as well. Finally, it fixes a few key issues on Windows, so if you're targeting Windows as a platform, it's almost mandatory to update. If you're only targeting iOS/Android, then upgrading is still recommended as we won't be backporting bug fixes to the 4.x line of releases.

Of course, if you have never had issues with Realm and you don't plan to take advantage of any of the new features, then probably it doesn't matter much and you can just keep using 4.x.

**Reply by rodion-m** [reply]
@nirinchev I got it. Thank you very much for a great support!

**Reply by nirinchev** [reply]
We're always here to help ❤️
